scrabbler/simulation.py

```diff
@@ -105,25 +111,19 @@ def make_move(self, move):
         if move.direction == "across":
             for i in range(start_column, start_column + len(move.word)):
                 if self.board.square(start_row, i).tile is None:
-                    self.remove_tile_from_rack(move, i - start_column)
+                    self.players[self.player].remove_tile_from_rack(move, i - start_column)
         else:
             for i in range(start_row, start_row + len(move.word)):
                 if self.board.square(i, start_column).tile is None:
-                    self.remove_tile_from_rack(move, i - start_row)
+                    self.players[self.player].remove_tile_from_rack(move, i - start_row)
 
         # Actually play the move here
         self.game.play(move.start_square, move.word, move.direction)
         print("Player %d plays: %s" % (self.player + 1, move.word))
-        self.scores[self.player] += move.score
-
-    def remove_tile_from_rack(self, move, index):
-        # if letter is not in word then it must be because a ? was used to make letter
-        if move.word[index] not in self.racks[self.player]:
-            self.racks[self.player].remove('?')
-        else:
-            self.racks[self.player].remove(move.word[index])
+        self.players[self.player].update_score(move.score)
 
     def print_end_game_message(self):
-        print('\nGAME OVER!')
-        print("PLAYER 1 SCORE: %d ...... PLAYER 2 SCORE: %d" % (self.scores[0], self.scores[1]))
-        print('Average move-generation time:', sum(self.times) / len(self.times))
+        print("\nGAME OVER!")
+        print("PLAYER 1 SCORE: %d ...... PLAYER 2 SCORE: %d" % (self.players[0].get_score(), self.players[1].get_score()))
+        #commented this out because i was getting a divide by zero error
```

Comment:
I think we can probably remove all of the `self.times` usage at this point anyways.